core/ResManager/plAgeInfo.cpp

```diff
@@ -62,25 +67,28 @@ void plAgeInfo::readFromFile(const ST::string& filename) {
             addPage(page);
         }
     }
-
-    delete S;
 }
 
-void plAgeInfo::writeToFile(const ST::string& filename, PlasmaVer ver) {
-    hsStream* S;
+void plAgeInfo::writeToFile(const ST::string& filename, PlasmaVer ver) const {
     if (ver.isUniversal()) {
-        S = new hsFileStream();
-        ((hsFileStream*)S)->open(filename, fmCreate);
+        hsFileStream S;
+        S.open(filename, fmCreate);
+        writeToStream(&S, ver);
+        S.close();
     } else {
-        S = new plEncryptedStream();
+        plEncryptedStream S;
         plEncryptedStream::EncryptionType eType = plEncryptedStream::kEncAuto;
         if (ver.isNewPlasma())
             eType = plEncryptedStream::kEncAES;
         else
             eType = plEncryptedStream::kEncXtea;
-        ((plEncryptedStream*)S)->open(filename, fmCreate, eType);
+        S.open(filename, fmCreate, eType);
+        writeToStream(&S, ver);
+        S.close();
     }
+}
 
+void plAgeInfo::writeToStream(hsStream* S, PlasmaVer) const {
```

Comment:
The `PlasmaVer` parameter is unused here -- it should probably be removed.

Reply:
Done